### PR TITLE
Compression

### PR DESCRIPTION
**Pull request by deadtrickster**
Negotiate encoding, support gzip and deflate.
New metric - telemetry_scrape_encoded_size_bytes.

#15

### CODE REVIEW

**Comment by deadtrickster** [comment]
ping

**Comment by deadtrickster** [comment]
ping :-)

File: src/elli_prometheus.erl [hunk 1]
```diff
@@ -147,7 +147,7 @@ handle_full_response(Type, [Req, Code, _Hs, _B, {Timings, Sizes}], _Config) ->
   TypedLabels = case Type of
                   request_complete -> ["full" | Labels];
                   chunk_complete -> ["chunks" | Labels] %;
-                  %% _ -> Labels
+                                    %% _ -> Labels
```
Comment (yurrriq) [inline src/elli_prometheus.erl]: Could you drop this change?

**Review by yurrriq — CHANGES_REQUESTED** [review]
Other than the silly whitespace comment. This looks fine to me.

**Comment by deadtrickster** [comment]
yay!

**Pull request merged**